x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx

```diff
@@ -176,27 +183,30 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
       }
     },
   });
+  const { onOpenTimeline, ...ruleFromTimelineData } = useRuleFromTimeline();
+
+  useEffect(() => {
+    if (ruleFromTimelineData.updated) {
+      setFieldValue('index', ruleFromTimelineData.index);
+      setFieldValue('queryBar', ruleFromTimelineData.queryBar);
+    }
+  }, [ruleFromTimelineData, setFieldValue]);
+
   const {
-    index: formIndex,
-    ruleType: formRuleType,
-    queryBar: formQuery,
-    dataViewId: formDataViewId,
-    threatIndex: formThreatIndex,
+    index,
+    ruleType,
+    queryBar,
+    dataViewId,
+    threatIndex,
     threatMapping: formThreatMapping,
-    machineLearningJobId: formMachineLearningJobId,
-    dataSourceType: formDataSourceType,
-    newTermsFields: formNewTermsFields,
+    machineLearningJobId,
+    dataSourceType,
+    newTermsFields,
     shouldLoadQueryDynamically: formShouldLoadQueryDynamically,
   } = formData;
 
   const [isQueryBarValid, setIsQueryBarValid] = useState(false);
   const [isThreatQueryBarValid, setIsThreatQueryBarValid] = useState(false);
-  const index = formIndex || initialState.index;
```

Comment:
not sure why it was like this because `formData` is initially set from `initialState` as an argument to `useForm`